src/main/java/Duke.java

```diff
@@ -1,10 +1,154 @@
+                        tasks.get(taskIndex).markAsNotDone();
+                        System.out.println("OK, I've marked this task as not done yet:");
+                        System.out.println(tasks.get(taskIndex));
+                    }
+                }
+                catch (EkudException error) {
+                    System.out.println("The task number is not valid or not provided.");
+                }
+            }
+            else if(userInputWords[0].equals("todo")){
+                int dividerPosition = userInput.indexOf(" ");
+                try {
+                    if (dividerPosition == -1) {
+                        throw new EkudException();
+                    }
+                    else {
+                        tasks.add(new Todo(userInput.substring(dividerPosition + 1)));
+                        System.out.println("Got it. I've added this task:");
+                        System.out.println(tasks.get(taskCount));
+                        taskCount++;
+                        System.out.println("Now you have " + taskCount + " tasks in the list.");
+                    }
+                }
+                catch (EkudException error) {
+                    System.out.println("The description of a todo cannot be empty.");
+                }
+            }
+            else if(userInputWords[0].equals("deadline")){
+                int dividerPosition = userInput.indexOf(" ");
+                try {
+                    if (dividerPosition == -1) {
+                        throw new EkudException();
+                    }
+                    else {
+                        int slashPosition = userInput.indexOf("/by");
+                        tasks.add(new Deadline(userInput.substring(dividerPosition + 1, slashPosition - 1), userInput.substring(slashPosition + 4)));
+                        System.out.println("Got it. I've added this task:");
+                        System.out.println(tasks.get(taskCount));
+                        taskCount++;
+                        System.out.println("Now you have " + taskCount + " tasks in the list.");
+                    }
+                }
+                catch (EkudException error) {
+                    System.out.println("The description of a deadline cannot be empty.");
+                }
+            }
+            else if(userInputWords[0].equals("event")){
+                int dividerPosition = userInput.indexOf(" ");
+                try {
+                    if (dividerPosition == -1) {
+                        throw new EkudException();
+                    }
+                    else {
+                        int fromPosition = userInput.indexOf("/from");
+                        int toPosition = userInput.indexOf("/to");
+
+                        tasks.add(new Event(userInput.substring(dividerPosition + 1, fromPosition - 1), userInput.substring(fromPosition + 6, toPosition - 1), userInput.substring(toPosition + 4)));
```

Comment:
This is pretty hard to read, especially with all the magic literals. Do refactor your magic literals. You can consider breaking this up into a function!